tests/integrations/ethereum/util/mock_rpc.go

```diff
@@ -38,7 +38,15 @@ func methodSigHash(sig string) string {
 }
 
 func responseErr(response serverResponse, err interface{}) serverResponse {
-	response.Error = err
+	response.Error = struct {
+		Code    int         `json:"code"`
+		Message string      `json:"message"`
+		Data    interface{} `json:"data,omitempty"`
+	}{
+		Code: 999,
+		Message: fmt.Sprintf("responding with err: %v", err),
```

Comment:
seems redundant to have a message that's the exact same as data, when this struct is used a single time

Reply:
Just having the string is incompatible with go-ethereum's internal rpc error type, which expects a structure like this. If an error happens and it isn't parsed, it errors out with a failure to decode it, ignoring the content.